feat: initial setup for Sauce Visual (#2486)
saucelabs/sauce-docs: docs/visual-testing/faq.md

> ---

Reviewer: Each answer in FAQ has a bigger font - I think we can make the font smaller

id: faq
title: Sauce Visual Testing FAQ
sidebar_label: FAQ
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> ### Is Sauce Visual's image comparison pixel-based or DOM-based?

Reviewer: ```suggestion
## Is Sauce Visual's image comparison pixel-based or DOM-based?
```

Author: We set it to h3 because otherwise the question would become to big

The current implementation is Pixel-based only.

> ### Does Sauce Visual support "ignore regions"?

Reviewer: ```suggestion
## Does Sauce Visual support "ignore regions"?
```

Yes, "ignore regions" are supported. Currently, you can define these regions in code by specifying the locator of a Web Element or by setting the coordinates of a rectangle.

> ```java

Author: Can we 
- or show an example of both options
- or refer to the docs for each binding on how to do that

  // This rectangular area would be ignored, no  pixel
  // changes will be diffed in this zone.
  IgnoreRegion ignoreRegion = new IgnoreRegion(200,200,100,100);
```

> ### Can I integrate this into CI/CD testing?

Reviewer: ```suggestion
## Can I integrate this into CI/CD testing?
```

Author: We set it to h3 because otherwise the question would become to big

Yes, Sauce Visual is 100% API-based and natively utilizes the same tools as the rest of the Sauce product line. When integrating Visual testing into your scripts, you can execute them just like any other scripts.

> ### What languages/frameworks will be supported at launch?

Reviewer: ```suggestion
## What languages/frameworks will be supported at launch?
```

Author: We set it to h3 because otherwise the question would become to big

> Sauce Visual will support the following languages/frameworks at launch: Selenium/Appium/Cypress, WebDriverIO, Java JUnit/TestNG, and JavaScript. You can find examples of some of these at [https://github.com/saucelabs/visual-examples](https://github.com/saucelabs/visual-examples).

Author: ```suggestion
Sauce Visual will support Desktop/Mobile Web the following languages/frameworks at launch:

- [Selenium for Java JUnit/TestNG](./integrations/java.md)
- [Cypress](./integrations/cypress.md)
- [Storybook (In BETA)](./integrations/storybook.md)
- [WebDriverIO](./integrations/webdriverio.md)

You can find examples of some of these at [https://github.com/saucelabs/visual-examples](https://github.com/saucelabs/visual-examples).
```

> ### What subsequent languages/frameworks will be supported?

Reviewer: ```suggestion
## What subsequent languages/frameworks will be supported?
```

Author: We set it to h3 because otherwise the question would become to big

Support for additional languages/frameworks will be determined based on demand. Frameworks like Playwright, TestCafe, XCUIT, Espresso, etc., are currently under consideration. Contributions are encouraged. Check out the [examples project](https://github.com/saucelabs/visual-examples). If you create your own public library or binding, feel free to add another example to that project.

> ### When will it support "my favorite framework"?

Reviewer: ```suggestion
## When will it support "my favorite framework"?
```

Author: We set it to h3 because otherwise the question would become to big
